pkg/kubectl/cmd/apply_view_last_applied.go

```diff
@@ -0,0 +1,157 @@
+	FilenameOptions              resource.FilenameOptions
+	Selector                     string
+	LastAppliedConfigurationList []string
+	OutputFormat                 string
+	Factory                      cmdutil.Factory
+	Out                          io.Writer
+	ErrOut                       io.Writer
+}
+
+var (
+	applyViewLastAppliedLong = templates.LongDesc(`
+		View the latest last-applied-configuration annotations by type/name or file.
+
+		You can use --output jsonpath={...} to extract specific values using a jsonpath expression.`)
+
+	applyViewLastAppliedExample = templates.Examples(`
+		# View the latest applied configuration file for a resource by type/name.
+		# This is retrieved from the last-applied-configuration annotation on the resource
+		kubectl apply view-last-applied deployment/nginx
+
+		# View the last-applied-configuration annotations by file
+		kubectl apply view-last-applied -f deployment_nginx.yaml -o json`)
+)
+
+func NewCmdApplyViewLastApplied(f cmdutil.Factory, out, err io.Writer) *cobra.Command {
+	options := &ViewLastApplied{Out: out, ErrOut: err}
+	cmd := &cobra.Command{
+		Use:     "view-last-applied (TYPE [NAME | -l label] | TYPE/NAME | -f FILENAME)",
+		Short:   "View latest last-applied-configuration annotations of a resource/object",
+		Long:    applyViewLastAppliedLong,
+		Example: applyViewLastAppliedExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(f, args))
+			cmdutil.CheckErr(options.Validate(cmd))
+			cmdutil.CheckErr(options.RunApplyViewLastApplied())
+		},
+	}
+
+	cmdutil.AddOutputVarFlagsForMutation(cmd, &options.OutputFormat)
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", "", "Selector (label query) to filter on, supports '=', '==', and '!='.")
+	usage := "that contains the last-applied-configuration annotations"
+	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, usage)
+
+	return cmd
+}
+
+func (o *ViewLastApplied) Complete(f cmdutil.Factory, args []string) error {
+	mapper, typer, err := f.UnstructuredObject()
+	if err != nil {
+		return err
+	}
+
+	cmdNamespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	r := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.UnstructuredClientForMapping), unstructured.UnstructuredJSONScheme).
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(enforceNamespace, &o.FilenameOptions).
+		ResourceTypeOrNameArgs(enforceNamespace, args...).
+		SelectorParam(o.Selector).
+		Latest().
+		Flatten().
+		Do()
+	err = r.Err()
+	if err != nil {
+		return err
+	}
+
+	err = r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+
+		configString, err := kubectl.GetOriginalConfiguration(info.Mapping, info.Object)
+		if err != nil {
+			return err
+		}
+		if configString == nil {
+			return fmt.Errorf("no last-applied-configuration annotation found on resource: %s", info.Name)
```

Comment:
I'm not sure should I use `return fmt.Errorf(..` or `fmt.Fprintf(o.errOut..`, but when you dealing a branch of file, directly `return fm.Errorf` will not continue and exit I think.